src/meta-srv/src/election/etcd.rs

```diff
@@ -42,20 +48,50 @@ impl EtcdElection {
             .await
             .context(error::ConnectEtcdSnafu)?;
 
-        Self::with_etcd_client(leader_value, client)
+        Self::with_etcd_client(leader_value, client).await
     }
 
-    pub fn with_etcd_client<E>(leader_value: E, client: Client) -> Result<ElectionRef>
+    pub async fn with_etcd_client<E>(leader_value: E, client: Client) -> Result<ElectionRef>
     where
         E: AsRef<str>,
     {
-        let leader_value = leader_value.as_ref().into();
+        let leader_value: String = leader_value.as_ref().into();
+
+        let leader_ident = leader_value.clone();
+        let (tx, mut rx) = broadcast::channel(1024);
```

Comment:
question: 1024 may be too large.

Reply:
Well, I think it's tolerable, because the elements it holds are `enum LeaderChangeMessage` with box pointers, not too large. That said, I reduce the capacity to 100.